app/models/chef.rb

```diff
@@ -1,4 +1,10 @@
 class Chef < ApplicationRecord
    validates_presence_of :name
    has_many :dishes
+   has_many :dish_ingredients, through: :dishes
+   has_many :ingredients, through: :dish_ingredients
```

Comment:
This looks great, but because there's already an association between ingredients and dishes, you can actually consolidate this down in: `has_many :ingredients, through: :dishes`